File: code/swift/rest/server/Sources/App/Models/Task.swift
import Fluent
import Foundation
import Vapor

enum TaskStatus: String, Codable, CaseIterable {
    case pending = "PENDING"
    case inProgress = "IN_PROGRESS"
    case completed = "COMPLETED"
    case cancelled = "CANCELLED"
    case archived = "ARCHIVED"
}

enum TaskPriority: String, Codable, CaseIterable {
    case low = "LOW"
    case medium = "MEDIUM"
    case high = "HIGH"
    case critical = "CRITICAL"
}

final class Task: Model, Content {
    static let schema = "tasks"

    @ID(key: .id)
    var id: UUID?

    @Field(key: "title")
    var title: String

    @OptionalField(key: "description")
    var description: String?

    @Enum(key: "status")
    var status: TaskStatus

    @Enum(key: "priority")
    var priority: TaskPriority

    @Field(key: "tags")
    var tags: [String]

    @OptionalField(key: "assigned_to")
    var assignedTo: String?

    @OptionalField(key: "due_date")
    var dueDate: Date?

    @Timestamp(key: "created_at", on: .create)
    var createdAt: Date?

    @Timestamp(key: "updated_at", on: .update)
    var updatedAt: Date?

    @OptionalField(key: "created_by")
    var createdBy: String?

    @OptionalField(key: "updated_by")
    var updatedBy: String?

    init() { }

    init(id: UUID? = nil,
         title: String,
         description: String? = nil,
         status: TaskStatus = .pending,
         priority: TaskPriority = .medium,
         tags: [String] = [],
         assignedTo: String? = nil,
         dueDate: Date? = nil,
         createdBy: String? = nil,
         updatedBy: String? = nil) {
        self.id = id
        self.title = title
        self.description = description
        self.status = status
        self.priority = priority
        self.tags = tags
        self.assignedTo = assignedTo
        self.dueDate = dueDate
        self.createdBy = createdBy
        self.updatedBy = updatedBy
    }
}

struct CreateTaskRequest: Content {
    let title: String
    let description: String?
    let priority: TaskPriority?
    let tags: [String]?
    let assignedTo: String?
    let dueDate: Date?
}

struct UpdateTaskRequest: Content {
    let title: String?
    let description: String?
    let status: TaskStatus?
    let priority: TaskPriority?
    let tags: [String]?
    let assignedTo: String?
    let dueDate: Date?
}

struct UpdateStatusRequest: Content {
    let status: TaskStatus
}

struct TasksResponse: Content {
    let items: [Task]
    let pageSize: Int
    let nextPageToken: String?
    let totalCount: Int
}

struct ErrorResponse: Content {
    let error: String
    let message: String
}

File: code/csharp/rest/client/Models/Task.cs
using System.Text.Json.Serialization;

namespace TaskRestClient.Models;

[JsonConverter(typeof(JsonStringEnumConverter))]
public enum TaskStatus
{
    Pending,
    InProgress,
    Completed
}

[JsonConverter(typeof(JsonStringEnumConverter))]
public enum TaskPriority
{
    Low,
    Medium,
    High,
    Critical
}

public class Task
{
    [JsonPropertyName("id")]
    public string Id { get; set; } = string.Empty;
    
    [JsonPropertyName("title")]
    public string Title { get; set; } = string.Empty;
    
    [JsonPropertyName("description")]
    public string? Description { get; set; }
    
    [JsonPropertyName("status")]
    public TaskStatus Status { get; set; } = TaskStatus.Pending;
    
    [JsonPropertyName("priority")]
    public TaskPriority Priority { get; set; } = TaskPriority.Medium;
    
    [JsonPropertyName("tags")]
    public List<string> Tags { get; set; } = new();
    
    [JsonPropertyName("assigned_to")]
    public string? AssignedTo { get; set; }
    
    [JsonPropertyName("created_by")]
    public string CreatedBy { get; set; } = string.Empty;
    
    [JsonPropertyName("created_at")]
    public DateTime CreatedAt { get; set; }
    
    [JsonPropertyName("updated_at")]
    public DateTime UpdatedAt { get; set; }
    
    [JsonPropertyName("due_date")]
    public DateTime? DueDate { get; set; }
}

public class CreateTaskRequest
{
    [JsonPropertyName("title")]
    public string Title { get; set; } = string.Empty;
    
    [JsonPropertyName("description")]
    public string? Description { get; set; }
    
    [JsonPropertyName("priority")]
    public TaskPriority Priority { get; set; } = TaskPriority.Medium;
    
    [JsonPropertyName("tags")]
    public List<string>? Tags { get; set; }
    
    [JsonPropertyName("assigned_to")]
    public string? AssignedTo { get; set; }
    
    [JsonPropertyName("created_by")]
    public string? CreatedBy { get; set; }
    
    [JsonPropertyName("due_date")]
    public DateTime? DueDate { get; set; }
}

public class UpdateTaskRequest
{
    [JsonPropertyName("title")]
    public string? Title { get; set; }
    
    [JsonPropertyName("description")]
    public string? Description { get; set; }
    
    [JsonPropertyName("status")]
    public TaskStatus? Status { get; set; }
    
    [JsonPropertyName("priority")]
    public TaskPriority? Priority { get; set; }
    
    [JsonPropertyName("tags")]
    public List<string>? Tags { get; set; }
    
    [JsonPropertyName("assigned_to")]
    public string? AssignedTo { get; set; }
    
    [JsonPropertyName("due_date")]
    public DateTime? DueDate { get; set; }
}

public class ListTasksResponse
{
    [JsonPropertyName("tasks")]
    public List<Task> Tasks { get; set; } = new();
    
    [JsonPropertyName("next_page_token")]
    public string? NextPageToken { get; set; }
    
    [JsonPropertyName("total_count")]
    public int TotalCount { get; set; }
}

public class ErrorResponse
{
    [JsonPropertyName("message")]
    public string Message { get; set; } = string.Empty;
    
    [JsonPropertyName("errors")]
    public Dictionary<string, List<string>>? Errors { get; set; }
}